components/service/glean/src/main/java/mozilla/components/service/glean/storages/EventsStorageEngine.kt

```diff
@@ -169,15 +334,29 @@ internal object EventsStorageEngine : StorageEngine {
 
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     override fun clearAllStores() {
+        storageDirectory.listFiles()?.forEach {
```

Comment:
Does this `forEach` need to run in the `GlobalScope.launch(KotlinDispatchers.IO) ` context? Otherwise we risk racing with writes, right?

Reply:
Yes, potentially.  However, this is should probably also block the main thread until that work is done.  That should be fine since this is a testing-only function.